release/sil/sil.bcc-latn.upp_ptwl1/source/sil.bcc-latn.upp_ptwl1.model.ts

```diff
@@ -10,5 +10,36 @@ const source: LexicalModelSource = {
   format: 'trie-1.0',
   wordBreaker: 'default',
   sources: ['wordlist.tsv'],
+  searchTermToKey: function (term) {
+    // Use this pattern to remove common diacritical marks.
+    // See: https://www.compart.com/en/unicode/block/U+0300
+    const COMBINING_DIACRITICAL_MARKS = /[\u0300-\u036f]/g;
+   
+    // Converts to Unicode Normalization form D.
+    // This means that MOST accents and diacritics have been "decomposed" and
+    // are stored as separate characters. We can then remove these separate
+    // characters!
+    //
+    // e.g., � ? A + �
+    let normalizedTerm = term.normalize('NFD');
+   
+    // Now, make it lowercase.
+    //
+    // e.g.,  A + � ? a + �
+    let lowercasedTerm = normalizedTerm.toLowerCase();
+   
+    // Now, using the pattern above replace each accent and diacritic with the
+    // empty string. This effectively removes all accents and diacritics!
+    //
+    // e.g.,  a + � ? a
+    //
+    // In Balochi we don't want to convert accented letters to their base form.
+    // each accented letter should be treated as an equal letter.
+    //  let termWithoutDiacritics = lowercasedTerm.replace(COMBINING_DIACRITICAL_MARKS, '')
```

Comment:
The reason `searchTermToKey()` exists is so that the typist can be sloppy, and the keyboard will autocorrect them and add the appropriate diacritics later. `searchTermToKey()` facilitates this usage: essentially, it defines what the "sloppy" version of the text is, and as long as the wordlist is correct (has correct spelling, correct diacritics), then the predicted text will be correct as well.

I know nothing of Balochi orthography, but is there any case when a typist can be sloppy (e.g., missing diacritics?) but they will still be understood?